internal/tiltfile/tiltfile_state.go

```diff
@@ -633,6 +672,20 @@ func (s *tiltfileState) assembleK8sV2() error {
 	return nil
 }
 
+func (s *tiltfileState) addEntityToResourceAndRemoveFromUnresourced(e k8s.K8sEntity, r *k8sResource) error {
+	r.entities = append(r.entities, e)
+	for i, ur := range s.k8sUnresourced {
+		if ur == e {
+			// delete from unresourced
+			s.k8sUnresourced = append(s.k8sUnresourced[:i], s.k8sUnresourced[i+1:]...)
+			return nil
+
+		}
+	}
+
+	return fmt.Errorf("Unable to find %s in unresourced YAML", e.Name())
```

Comment:
if we hit this, it's a bug, right?
maybe worth specifying:
1. that it's an internal error
2. the name of the resource on which we encountered it

Reply:
We'll hit this if two resources try to include the same object, right? This seems like a confusing error to get in that case.